src/test/ui/deprecation/derive_on_deprecated_forbidden.rs

```diff
@@ -0,0 +1,7 @@
+#![forbid(deprecated)]
+
+#[deprecated = "oh no"]
+#[derive(Default)]
+struct X; //~ ERROR deprecated
```

Comment:
Note that erroring on `forbid` is a regression compared to stable. https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c5255de60042c80607e6597f901295d8

Reply:
I consider that a bugfix. If you are using forbid, you don't want any occurrances of a lint to go unnoticed.

Reply:
OK, for the example here it's still just noise. Maybe I can just not emit deprecation warnings inside derive expansions

Reply:
I suppose that's true :)